text/204-golang-bindings.md

```diff
@@ -0,0 +1,655 @@
+  (All the datatype interfaces in the [JSII calc test assembly](https://github.com/aws/jsii/blob/master/packages/jsii-calc/test/assembly.jsii) appear
+to only have immutable, i.e. readonly, properties).
+
+## Classes
+
+Typescript takes an object-oriented approach to classes, which includes using polymorphism and subtyping, neither of which are natively supported in
+Go, which is not an object-oriented language. While custom structs, which can be used as pointer receivers in function definitions to simulate
+methods", can be used to encapsulate object-like behavior in Go, subtyping on these custom structs is not possible. In order to simulate subtyping, we
+would need to generate an interface in addition to a concrete struct for each JSII class.
+
+The JSII [ClassType](https://github.com/aws/jsii/blob/master/packages/%40jsii/spec/lib/assembly.ts#L803) assembly provides information on whether a
+class is abstract, whether it extends another class, and whether it implements other interfaces. We will discuss each case in turn.
+
+### Case 1: Simple class
+
+```ts
+class Greeter {
+    greeting: string;
+
+    constructor(message: string) {
+        this.greeting = message;
+    }
+
+    greet(): string {
+        return "Hello, " + this.greeting;
+    }
+}
+
+let greeter = new Greeter("world");
+greeter.greet() // "Hello, world"
+```
+
+In Go: ([Go playground example](https://play.golang.org/p/cMW3Rv0lERh))
+
+```go
+package greeter
+
+type GreeterIface interface {
+    Greet() string
+}
+
+type Greeter struct {
+    greeting string
+}
+
+func New(message string) *Greeter {
+    return &Greeter{message}
+}
+func (g *Greeter) Greet() string {
+    return fmt.Sprintf("Hello, %+s", g.greeting)
+}
+
+// usage
+g := greeter.New("world")
+fmt.Println(g.Greet()) // "Hello, world"
+```
+
+### Case 2: Extending a Base class
+
+```ts
+class Animal {
+    name: string;
+    constructor(theName: string) { this.name = theName; }
+    move(distanceInMeters: number = 0) {
+        console.log(`${this.name} moved ${distanceInMeters}m.`);
+    }
+}
+
+class Snake extends Animal {
+    constructor(name: string) { super(name); }
+    move(distanceInMeters = 5) {
+        console.log("Slithering...");
+        super.move(distanceInMeters);
+    }
+}
+
+let sam = new Snake("Sammy the Python");
+
+sam.move();
+//  Slithering...
+// Sammy the Python moved 10m.
+```
+
+In Go:
+([Go playground example](https://play.golang.org/p/ZwoKmLn5OkV))
+([Example with embedded field as named property](https://play.golang.org/p/nWfmHhxbs1G))
+
+```go
+package animal
+
+// Base class as interface
+type AnimalIface interface {
+    Name() string
+    Move(distance int64)
+}
+
+// Base class implementation
+type Animal struct {
+    name string
+}
+
+func NewAnimal(name string) AnimalIface {
+    return &Animal{name}
+}
+
+func (a *Animal) Name() string {
```

Comment:
`name` is read/write. Where is the setter?

Reply:
changed this to be just a getter for purposes of the example, which is meant to focus on the inheritence chain and method overriding.